evm/src/cpu/cpu_stark.rs

```diff
@@ -61,16 +62,42 @@ pub fn ctl_filter_logic<F: Field>() -> Column<F> {
 }
 
 pub fn ctl_data_memory<F: Field>(channel: usize) -> Vec<Column<F>> {
-    debug_assert!(channel < NUM_CHANNELS);
-    let channel_map = COL_MAP.mem_channels[channel];
-    let mut cols: Vec<Column<F>> = Column::singles([
-        channel_map.is_read,
-        channel_map.addr_context,
-        channel_map.addr_segment,
-        channel_map.addr_virtual,
-    ])
-    .collect_vec();
-    cols.extend(Column::singles(channel_map.value));
+    let mut cols = Vec::<Column<F>>::new();
+    if channel == membus::channel_indices::CODE {
```

Comment:
I think it may be a little nicer to split this into two functions, for the code channel and `i`th GP channel. Then we wouldn't need to adjust the index, the caller would just enumerate `0..NUM_GP_CHANNELS`.